frontend_tests/zjsunit/parallel_runner.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+const child_process = require("child_process");
+
+const parallel = process.argv[2];
+
+// Find the files we need to run.
+const files = process.argv.slice(3);
+if (files.length === 0) {
+    throw new Error("No tests found");
+}
+
+const index = require.resolve("./index");
+
+function create_file_groups(files, parallel) {
+    const group_size = Math.ceil(files.length / parallel);
+    const file_groups = [];
+    for (let i = 0; i < parallel; i += 1) {
+        const file_group = files.slice(i * group_size, (i + 1) * group_size);
+        // never generate an empty group or send it to a forked process.
+        if (file_group.length > 0) {
+            file_groups.push(file_group);
+        }
+    }
+    return file_groups;
+}
+
+const file_groups = create_file_groups(files, parallel);
+
+const worker_processes = [];
+
+// we use this map to ensure that we only call .kill() on the processes
+// we have forked, this is important because if we didn't ensure this,
+// it is entirely possible that we would send a kill signal to an
+// arbitrary process which just happens to have the same pid as the one
+// we had created (through pid reallocation).
+// see: https://nodejs.org/api/child_process.html#subprocesskillsignal
+const has_exited = new Map();
```

Comment:
Not necessary. Node already keeps track of which processes have exited and avoids signaling them. It won’t be fooled by PID reallocation.

Reply:
> Sending a signal to a child process that has already exited is not an error but may have unforeseen consequences. Specifically, if the process identifier (PID) has been reassigned to another process, the signal will be delivered to that process instead which can have unexpected results.

The above is what the doc I linked says, there are also one or two SO answers that pointed me to this possibility, do you believe the docs are outdated? If so, then do you have a source that you / I could use to ask them to update?